### PR TITLE
feat: add back to top component

### PR DESCRIPTION
**Pull request by maomaoZH**

### CODE REVIEW

**Comment by maomaoZH** [comment]
sry @eeegor  still need to update css class name. not ready for review yet.

**Comment by maomaoZH** [comment]
ready to go. thanks

File: packages/components/src/components/back-to-top/back-to-top.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import { CssClassMap } from '../../utils/utils';
+import classNames from 'classnames';
+
+@Component({
+  tag: 't-back-to-top',
+  styleUrl: 'back-to-top.css',
+  shadow: true,
+})
+export class BackToTop {
+  @Prop() public visibilityHeight?: number = 200;
```
Comment (eeegor) [inline packages/components/src/components/back-to-top/back-to-top.tsx]: Please add documentation

File: packages/components/src/components/back-to-top/back-to-top.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  @State() private visible: boolean = false;
+
+  @Listen('scroll', { target: 'window' })
+  public handleScroll(ev) {
```
Comment (eeegor) [inline packages/components/src/components/back-to-top/back-to-top.tsx]: Please don't use abbreviations e.g. `ev`, use `event` instead

File: packages/components/src/components/back-to-top/back-to-top.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  public render() {
+    this.visible = document.documentElement.scrollTop > this.visibilityHeight;
+    return (
+      <div
+        class={this.getCssClassMap()}
+        onClick={this.scrollToTop}
+        onScroll={() => this.handleScroll}
+      >
+        <slot />
```
Comment (eeegor) [inline packages/components/src/components/back-to-top/back-to-top.tsx]: The slot is not very usable right now as the `css` is not compatible. If there is a slot, the arrow is still displayed and the scroll button is not adapting the slot content. 
image

**Comment by maomaoZH** [comment]
update code accordingly

**Pull request closed without merging**